spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java

```diff
@@ -186,6 +186,13 @@ public RabbitTemplate rabbitTemplate(ConnectionFactory connectionFactory) {
 			if (templateProperties.getReplyTimeout() != null) {
 				rabbitTemplate.setReplyTimeout(templateProperties.getReplyTimeout());
 			}
+			if (templateProperties.getExchange() != null) {
+				rabbitTemplate.setExchange(templateProperties.getExchange());
+			}
+			if (templateProperties.getRoutingKey() != null) {
+				rabbitTemplate.setRoutingKey(templateProperties.getRoutingKey());
+			}
+			rabbitTemplate.setChannelTransacted(templateProperties.isChannelTransacted());
```

Comment:
A test should cover this. If we go with the route of flagging `""` a the default value. We should have a test that this value is consistent with what happens  by default.